ion-schema/src/isl/isl_type_reference.rs

```diff
@@ -106,6 +107,19 @@ impl IslTypeRefImpl {
         })
     }
 
+    /// Verifies if the given type reference contains an `occurs` field or not
+    // This is used to make sure only `ordered_elements` and `fields` constraint can contain `occurs`.
+    // This method returns `false` for `Named` or `TypeImport` type references because `occurs` field is not allowed within named type definitions.
+    pub fn get_occurs_constraint(&self) -> bool {
+        match self {
+            IslTypeRefImpl::Anonymous(anonymous_type_def) => anonymous_type_def
+                .constraints()
+                .iter()
+                .any(|c| matches!(c, IslConstraintImpl::Occurs(_))),
+            _ => false,
+        }
+    }
```

Comment:
Does not need to be fixed in this PR.

This suggests to me that we should be modeling `VariablyOccuringTypeReference` as a distinct type from `TypeReference` and stop treating `occurs` as a constraint.

Reply:
Makes sense. I will create an issue for it.

Reply:
Reference issue: https://github.com/amazon-ion/ion-schema-rust/issues/157